File: app/src/main/java/com/the_roberto/kittengifs/giphy/GiphyService.java
package com.the_roberto.kittengifs.giphy;

import retrofit.Callback;
import retrofit.http.GET;
import retrofit.http.Query;

public interface GiphyService {

    @GET("/gifs/random")
    void getRandomGif(@Query("tag") String tag, Callback<RandomGifResponse> callback);

    @GET("/gifs/search")
    void getGifs(@Query("q") String query, @Query("offset") long offset, @Query("limit") long limit, Callback<SearchGifResponse> callback);
}

File: app/src/main/java/com/the_roberto/kittengifs/giphy/GiphyService.kt
package com.the_roberto.kittengifs.giphy

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface GiphyService {

    @GET("gifs/random")
    fun getRandomGif(@Query("tag") tag: String): Call<RandomGifResponse>

    @GET("gifs/search")
    fun getGifs(@Query("q") query: String, @Query("offset") offset: Long, @Query("limit") limit: Long): Call<SearchGifResponse>
}
